extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java

```diff
@@ -287,7 +288,7 @@ AdditionalBeanBuildItem additionalBeans() {
                         TemplateInstance.class, CollectionTemplateExtensions.class,
                         MapTemplateExtensions.class, NumberTemplateExtensions.class, ConfigTemplateExtensions.class,
                         TimeTemplateExtensions.class, StringTemplateExtensions.class, OrOperatorTemplateExtensions.class,
-                        ObjectsTemplateExtensions.class)
+                        ObjectsTemplateExtensions.class, DebugQuteEngineObserver.class)
```

Comment:
I think that we should only register the `DebugQuteEngineObserver` in the dev mode, i.e. register the bean in a build step annotated with `@BuildStep(onlyIf = IsLocalDevelopment.class)`.

Reply:
You mean that I need just to annotate `DebugQuteEngineObserver `with `@BuildStep(onlyIf = IsLocalDevelopment.class)` and remove `DebugQuteEngineObserver.class` ?

Reply:
If we do that it means that we can not debug Quarkus app in production mode?